### PR TITLE
Maintenance Guides: Basic link clean up in advance of translation of content

### PR DESCRIPTION
**Pull request by krisdale**
- Remove unused links
- Updated links  defined as named refs to use simple text e.g. name of linked content
instead of narrative text
- Minor edits to accomodate minor changes to name of ref links

Signed-off-by: Kristal Dale <kristal.dale@intel.com>

### CODE REVIEW

File: source/clear-linux/guides/maintenance/kernel-development.rst [hunk 1]
```diff
@@ -6,26 +6,28 @@ Kernel development
...
 .. contents::
    :local:
    :depth: 1
    :backlinks: top
 
-Source RPM files (SRPM) are also available for all |CL| kernels, and can be
-used for development instead. Select this link to view the latest `source RPM files`_.
+Source RPM (SRPM) files are also available for all |CL| kernels, and can be
+used for development instead. Select this link to view the latest
+`SRPM`_ files.
```
Comment (mvincerx) [inline source/clear-linux/guides/maintenance/kernel-development.rst]: [Index page](https://cdn.download.clearlinux.org/current/source/SRPMS/) uses "SRPMS", please add S to match it.

File: source/clear-linux/guides/maintenance/kernel-modules-dkms.rst [hunk 2]
```diff
@@ -315,20 +315,16 @@ Examples
...
 **************
 
-* `Dynamic Kernel Module System (DKMS) project on GitHub <https://github.com/dell/dkms>`_
+* `Dynamic Kernel Module System (DKMS)`_
 
 * `Dell Linux Engineering Dynamic Kernel Module Support: From Theory to Practice <https://www.kernel.org/doc/ols/2004/ols2004v1-pages-187-202.pdf>`_
 
 * `Linux Journal: Exploring Dynamic Kernel Module Support <https://www.linuxjournal.com/article/6896>`_
 
+.. _mixer: https://clearlinux.org/features/mixer-tool
```
Comment (mvincerx) [inline source/clear-linux/guides/maintenance/kernel-modules-dkms.rst]: Please update to :ref:`mixer` . Link used ref to deprecated feature page.

File: source/clear-linux/guides/maintenance/kernel-modules.rst [hunk 3]
```diff
@@ -216,6 +217,4 @@ Related topic
 
 * :ref:`kernel-modules-dkms`
 
-
-.. _`on GitHub`: https://github.com/clearlinux/distribution
 .. _`mixer tool`: https://clearlinux.org/features/mixer-tool
```
Comment (mvincerx) [inline source/clear-linux/guides/maintenance/kernel-modules.rst]: Please update to `:ref:mixer` on Ln 71 above.  Delete hyperlink at bottom. Thanks

File: source/clear-linux/guides/maintenance/kernel-modules-dkms.rst (hunk 2 above)
Comment (krisdale) [reply source/clear-linux/guides/maintenance/kernel-modules-dkms.rst]: Nice catch!

File: source/clear-linux/guides/maintenance/kernel-development.rst (hunk 1 above)
Comment (krisdale) [reply source/clear-linux/guides/maintenance/kernel-development.rst]: Good catch - thanks :)

**Comment by krisdale** [comment]
@mvincerx Good catches - thanks for that! Changes applied.

**Comment by mvincerx** [comment]
LGTM

**Pull request merged**